### PR TITLE
Small corrections

### PR DESCRIPTION
**Pull request by oualib**
Small Correction
 - set_parameters
 - some tests
 - udf

### CODE REVIEW

File: verticapy/tests/vModel/test_normalizer.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+
+class TestNormalizer:
+
+    def test_deploySQL(self, model):
+        expected_sql = 'APPLY_NORMALIZE("citric_acid", "residual_sugar", "alcohol" USING PARAMETERS model_name = \'norm_model_test\', match_by_pos = \'true\')'
+        result_sql = model.deploySQL()
+
+        assert result_sql == expected_sql
+
```
Comment (afard) [inline verticapy/tests/vModel/test_normalizer.py:51]: Please add a `test_deployInverseSQL` as well.

File: verticapy/tests/vModel/test_pca.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    model_class.drop()
+
+class TestPCA:
+
+    def test_deploySQL(self, model):
+        expected_sql = 'APPLY_PCA("citric_acid", "residual_sugar", "alcohol" USING PARAMETERS model_name = \'pca_model_test\', match_by_pos = \'true\', cutoff = 1)'
+        result_sql = model.deploySQL()
+
+        assert result_sql == expected_sql
+
```
Comment (afard) [inline verticapy/tests/vModel/test_pca.py:50]: Please add a test_deployInverseSQL as well.

File: verticapy/tests/vModel/test_normalizer.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+            "public.winequality", ["citric_acid", "residual_sugar", "alcohol"]
+        )
+        md = model2.to_sklearn()
+        model2.cursor.execute(
+            "SELECT APPLY_NORMALIZE(citric_acid, residual_sugar, alcohol USING PARAMETERS model_name = '{}', match_by_pos=True) FROM (SELECT 3.0 AS citric_acid, 11.0 AS residual_sugar, 93. AS alcohol) x".format(
+                model2.name
+            )
+        )
+        prediction = model2.cursor.fetchone()[0]
+        assert prediction == pytest.approx(md.transform([[3.0, 11.0, 93.0]])[0][0])
```
Comment (afard) [inline verticapy/tests/vModel/test_normalizer.py:119]: A good written test should clean up after itself.
Please drop model2 when you don't need it anymore.

File: verticapy/tests/vModel/test_normalizer.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+        )
+        md = model3.to_sklearn()
+        model3.cursor.execute(
+            "SELECT APPLY_NORMALIZE(citric_acid, residual_sugar, alcohol USING PARAMETERS model_name = '{}', match_by_pos=True) FROM (SELECT 3.0 AS citric_acid, 11.0 AS residual_sugar, 93. AS alcohol) x".format(
+                model3.name
+            )
+        )
+        prediction = model3.cursor.fetchone()[0]
+        assert prediction == pytest.approx(md.transform([[3.0, 11.0, 93.0]])[0][0])
+
```
Comment (afard) [inline verticapy/tests/vModel/test_normalizer.py:134]: Please add `model3.drop()` here

File: verticapy/tests/vModel/test_normalizer.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+        )
+        assert winequality_trans["citric_acid"].min() == pytest.approx(
+            0.0, abs=1e-6
+        )
+        assert winequality_trans["residual_sugar"].max() == pytest.approx(
+            1.0, abs=1e-6
+        )
+        assert winequality_trans["alcohol"].min() == pytest.approx(
+            0.0, abs=1e-6
+        )
```
Comment (afard) [inline verticapy/tests/vModel/test_normalizer.py:168]: Please add `model2.drop()` here to clean up the database.

File: verticapy/tests/vModel/test_normalizer.py [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+        assert winequality_trans["citric_acid"].median() == pytest.approx(
+            0.0, abs=1e-6
+        )
+        assert winequality_trans["residual_sugar"].median() == pytest.approx(
+            0.0, abs=1e-6
+        )
+        assert winequality_trans["alcohol"].median() == pytest.approx(
+            0.0, abs=1e-6
+        )
+
```
Comment (afard) [inline verticapy/tests/vModel/test_normalizer.py:188]: Please add `model3.drop()` here to clean up the database.

File: verticapy/tests/vModel/test_svd.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    model_class.drop()
+
+class TestSVD:
+
+    def test_deploySQL(self, model):
+        expected_sql = 'APPLY_SVD("citric_acid", "residual_sugar", "alcohol" USING PARAMETERS model_name = \'SVD_model_test\', match_by_pos = \'true\', cutoff = 1)'
+        result_sql = model.deploySQL()
+
+        assert result_sql == expected_sql
+
```
Comment (afard) [inline verticapy/tests/vModel/test_svd.py:50]: Please add test_deployInverseSQL as well.

File: verticapy/tests/vModel/test_svd.py [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+        assert m_att_details["value"][0] == pytest.approx(968.964362586858, abs=1e-6)
+        assert m_att_details["value"][1] == pytest.approx(354.585184720344, abs=1e-6)
+        assert m_att_details["value"][2] == pytest.approx(11.7281921567471, abs=1e-6)
+
+    def test_get_params(self, model):
+        assert model.get_params() == {'method': 'lapack', 'n_components': 0}
+
+    def test_to_sklearn(self, model):
+        # Zscore
```
Comment (afard) [inline verticapy/tests/vModel/test_svd.py:96]: The `# Zscore` comment looks irrelevant.

**Review by afard — CHANGES_REQUESTED** [review]
Looks good in general. I have left you a few comments to improve the tests.
It would be very nice if you can address them before I merge the PR.

File: verticapy/learn/vmodel.py [hunk 9]
```diff
@@ -576,6 +576,10 @@ def set_params(self, parameters: dict = {}):
 	parameters: dict, optional
 		New parameters.
 		"""
+        try:
+            self.parameters
```
[the remark this answers is not in the document]
Comment (oualib) [reply verticapy/learn/vmodel.py:580]: It could if the model has no attribute parameters yet.

File: verticapy/learn/vmodel.py [hunk 10]
```diff
@@ -1744,12 +1919,12 @@ def to_sklearn(self):
...
             for i in range(len(self.X)):
                 features[str_column(self.X[i]).lower()] = i
-            if isinstance(self, (vens.RandomForestRegressor,)):
+            if isinstance(self, (vens.RandomForestRegressor,)) or self.type == "RandomForestRegressor":
                 model = skens.RandomForestRegressor(
                     n_estimators=params["n_estimators"], **parameters
                 )
                 model.base_estimator_ = skdtree.DecisionTreeRegressor(**parameters)
-            elif isinstance(self, (vens.RandomForestClassifier,)):
+            elif isinstance(self, (vens.RandomForestClassifier,)) or self.type == "RandomForestClassifier":
```
[the remark this answers is not in the document]
Comment (oualib) [reply verticapy/learn/vmodel.py:1927]: Yes we can as this function is for all models. We can split it to smaller pieces and then assemble it

File: verticapy/learn/vmodel.py [hunk 11]
```diff
@@ -591,8 +595,10 @@ def set_params(self, parameters: dict = {}):
                     )
                 )
                 model_parameters["solver"] = parameters["solver"]
-            else:
+            elif "solver" not in self.parameters:
```
[the remark this answers is not in the document]
Comment (oualib) [reply verticapy/learn/vmodel.py:598]: It is the general set_params method. We are testing it in all models UT.

File: verticapy/tests/vModel/test_normalizer.py (hunk 1 above)
Comment (oualib) [reply verticapy/tests/vModel/test_normalizer.py:51]: done;

File: verticapy/tests/vModel/test_pca.py (hunk 2 above)
Comment (oualib) [reply verticapy/tests/vModel/test_pca.py:50]: Done;

File: verticapy/tests/vModel/test_normalizer.py (hunk 3 above)
Comment (oualib) [reply verticapy/tests/vModel/test_normalizer.py:119]: Done;

File: verticapy/tests/vModel/test_normalizer.py (hunk 4 above)
Comment (oualib) [reply verticapy/tests/vModel/test_normalizer.py:134]: Done;

File: verticapy/tests/vModel/test_normalizer.py (hunk 6 above)
Comment (oualib) [reply verticapy/tests/vModel/test_normalizer.py:188]: Done;

File: verticapy/tests/vModel/test_svd.py (hunk 7 above)
Comment (oualib) [reply verticapy/tests/vModel/test_svd.py:50]: Done;

File: verticapy/tests/vModel/test_svd.py (hunk 8 above)
Comment (oualib) [reply verticapy/tests/vModel/test_svd.py:96]: Erased!

**Review by afard — APPROVED** [review]
